Content.Server/VendingMachines/VendingMachineSystem.cs

```diff
@@ -435,6 +440,15 @@ public override void Update(float frameTime)
                     }
                 }
             }
+            var disabled = EntityQueryEnumerator<EmpDisabledComponent, VendingMachineComponent>();
+            while (disabled.MoveNext(out var uid, out _, out var comp))
+            {
+                if (comp.NextEmpEject < _timing.CurTime)
+                {
+                    EjectRandom(uid, true, false, comp);
+                    comp.NextEmpEject = _timing.CurTime + TimeSpan.FromSeconds(5 * comp.EjectDelay);
```

Comment:
Use += or else it gets rounding issues with timing.

Really this should be a separate component but I'm going to make a separate issue for the state of vending machine code.

Reply:
yes, but you would need to set up correct time in first iteration (maybe do it in OnEmpPulse handler)

Reply:
yeah that'll work